code/game/objects/structures/bonfire.dm

```diff
@@ -119,8 +119,21 @@
 
 /obj/structure/bonfire/proc/on_entered(datum/source, atom/movable/entered)
 	SIGNAL_HANDLER
-	if(burning & !grill)
-		bonfire_burn()
+	if(burning)
+		if(grill)
```

Comment:
```suggestion
		if(!grill)
```
Wait hold up

Reply:
motherFUCKER i was thinking of making it an early return but decided not to but didn't go back and change it an